manager/dispatcher/assignments.go

```diff
@@ -45,39 +34,85 @@ func newAssignmentSet(log *logrus.Entry, dp *drivers.DriverProvider) *assignment
 	}
 }
 
+func assignSecret(a *assignmentSet, readTx store.ReadTx, mapKey typeAndID) {
+	a.tasksUsingDependency[mapKey] = make(map[string]struct{})
+	secret, err := a.secret(readTx, mapKey.id)
+	if err != nil {
+		a.log.WithFields(logrus.Fields{
+			"resource.type": "secret",
+			"secret.id":     mapKey.id,
+			"error":         err,
+		}).Debug("failed to fetch secret")
+		return
+	}
+	if secret == nil {
```

Comment:
It might make sense to consolidate the errors, since it seems like the log fields are pretty similar and it might be good to log that the error is nil so we'd see that something had changed that wasn't intended to.